asf_heat_pump_suitability/pipeline/prepare_features/off_gas.py

```diff
@@ -0,0 +1,18 @@
+from asf_heat_pump_suitability.getters.get_datasets import get_df_spa_offgasgrid
+import polars as pl
+
+
+def add_off_gas_feature(df: pl.DataFrame) -> pl.DataFrame:
+    """
+    This function adds an 'OFF_GAS' column to a DataFrame based on whether 'POSTCODE' is in off_gas_postcodes.
+
+    Parameters:
+    df (pl.DataFrame): The EPC DataFrame to which the 'OFF_GAS' column will be added.
+
+    Returns:
+    pl.DataFrame: The DataFrame with the added 'OFF_GAS' column.
+    """
+    off_gas_df = get_df_spa_offgasgrid()
+    off_gas_postcodes = off_gas_df["Post Code"].str.replace(" ", "").to_list()
```

Comment:
```suggestion
    off_gas_postcodes = off_gas_df["Post Code"].str.replace(r'\s+', '').to_list()
```

I'm not sure why, but pandas seems to have an issue with the way you did it when there are double spaces. In this case of 'ZE3  9JZ' (two spaces) this code converts it to 'ZE3 9JZ' (one space). Quite confusing.. but the suggestion I gave seems to successfully remove all spaces:

![Screenshot 2024-07-26 at 10 08 24](https://github.com/user-attachments/assets/9ccae8e3-c617-4a7d-8b03-d61e54ffdbeb)

Reply:
should add another 455 properties as being off gas with this

Reply:
thanks @lizgzil this is a great spot - especially because I have written code in another file in this repo which may encounter the same problem.! I'll add an issue to fix

Reply:
Ah good spot @lizgzil! I'll make this change!